### PR TITLE
Fix mirror script when upstream set to newer version than downstream

### LINKED ISSUE #21690: Mirroring adds partial_implementation when version_added is false
### What type of issue is this?

Script issue

### What is the issue?

BCD exports this:

```
"hr_in_select": {
          "__compat": {
            "description": "<code>&lt;hr&gt;</code> in <code>&lt;select&gt;</code>",
            "source_file": "html/elements/hr.json",
            "support": {
              "samsunginternet_android": {
                "notes": "Exposes the <code>&lt;hr&gt;</code> as its own empty selectable option and not as a horizontal rule.",
                "partial_implementation": true,
                "version_added": false
              },
            }
          }
        },
```

### What behavior were you expecting?

I would have expected:

```
"hr_in_select": {
          "__compat": {
            "description": "<code>&lt;hr&gt;</code> in <code>&lt;select&gt;</code>",
            "source_file": "html/elements/hr.json",
            "support": {
              "samsunginternet_android": {
                "version_added": false
              },
            }
          }
        },
```

### What version(s) of BCD is the issue present in?

- [X] The current BCD release
- [X] The current version of the `main` branch

### Do you have anything more you want to share?

I think the schema/linter excludes partial_implementation and version_added at the same time? Are such checks also done against the built data?

Issue found by @dletorey. Thank you!

### PR DESCRIPTION
**Pull request by queengooborg**
This PR fixes the mirorring script to set the downstream browser data to `{version_added: false}` (AKA, don't copy notes and `partial_implementation`) if the upstream is set to a newer version than the latest downstream browser.  For example, if a feature was implemented in Chrome 121 and we're mirroring to Samsung Internet, but the latest version of Samsung Internet lines up with Chrome 115, it will be set to simply `false`.

Fixes #21690.

Note: some statements can now be set to mirror due to this change, so the lint is failing.  I will apply these changes once the PR is approved.

### CODE REVIEW

**Comment by Elchi3** [comment]
I think this PR adds the additional check to resolve https://github.com/mdn/browser-compat-data/issues/21690:

```js
if (newData.version_added === false && sourceData.version_added !== false) {
  // If the feature is added in an upstream version newer than available in the downstream browser, don't copy notes, etc.
  return { version_added: false };
}
```

I reviewed this bit by writing the following test code which I think we cannot add to the tests because the browser release data is a moving target and we don't have mocks for it:

```js
it('Dont copy notes etc. if feature is added in an upstream version newer than available in the downstream browser', () => {
  const support: InternalSupportBlock = {
    chrome_android: {
      version_added: '122',
      partial_implementation: true,
      notes: 'This feature has quirks.',
    },
  };

  const mirrored = mirrorSupport('samsunginternet_android', support);
  assert.deepEqual(mirrored, { version_added: false });
});
```
The test code fails on main and passes with this PR, so that seems good. (I don't know why some code was moved around, though)

Then there is something that this code also produces:

```json
"firefox": {
  "version_added": "preview",
  "partial_implementation": true,
  "notes": "Currently supported on Linux and Windows only."
},
"firefox_android": "mirror",
```

Instead of what we have now:

```json
"firefox": {
  "version_added": "preview",
  "partial_implementation": true,
  "notes": "Currently supported on Linux and Windows only."
},
"firefox_android": {
  "version_added": false
},
```

I think that's wrong. I think it shouldn't mirror. So, the CI failure is correct but I think the solution is not to change the data. It is to fix the mirror script. Maybe this problem is only due to "preview"?

**Review by Elchi3 — CHANGES_REQUESTED** [review]
See comment above

**Comment by queengooborg** [comment]
> we cannot add to the tests because the browser release data is a moving target and we don't have mocks for it

Yeah, we should be using mock browser data when running unittests in BCD.  That's what the collector does.

> I think that's wrong. I think it shouldn't mirror. So, the CI failure is correct but I think the solution is not to change the data. It is to fix the mirror script. Maybe this problem is only due to "preview"?

Hmm...this is a tricky one.  On one hand, I'd say that the mirroring script is outputting the correct results; if the downstream browser doesn't support preview editions, and the upstream browser is set to "preview", the downstream browser should indeed become "false".  On the other hand, if "preview" is replaced with a version number down the line, and the downstream browser is still set to mirror, the notes and partial implementation will get copied down.

Maybe this isn't much of an issue since we're running the collector on mobile browsers again?

**Comment by caugner** [comment]
FWIW Here is the impact of this change on the `data.json`:

https://github.com/mdn/browser-compat-data/blob/ce620083f1166fd4e2cff1165c30edfe9f4ab288/data.patch

PS: I ran `npm run build` on `main` and on this branch, committed the `/build` output to a git repo, and compared the two versions with `git diff --minimal HEAD^ -- data.json`.

File: api/GPU.json [hunk 1]
```diff
@@ -36,9 +36,7 @@
             "partial_implementation": true,
             "notes": "Currently supported on Linux and Windows only."
           },
-          "firefox_android": {
-            "version_added": false
-          },
+          "firefox_android": "mirror",
```
Comment (caugner) [inline api/GPU.json:39]: This does indeed not look right, as @Elchi3 spotted.

**Review by caugner — DISMISSED** [review]

File: api/GPU.json (hunk 1 above)
Comment (caugner) [reply api/GPU.json]: Reverted in 317b03784ccab89d43b0090d9227751ebdd0d66b.

**Comment by caugner** [comment]
@queengooborg Can you please take a look? This should be good to go.

**Review by caugner — COMMENTED** [review]
LGTM, but since I committed to this PR, I cannot give the final approval.

**Review by caugner — CHANGES_REQUESTED** [review]
Either something goes wrong during the merge, or this is broken. `npm run fix` sets `firefox_android` to `mirror` for those GPU features, although they are only supported on Linux and Windows.

**Comment by caugner** [comment]
> Either something went wrong during the merge, or this is broken. `npm run fix` sets `firefox_android` to `mirror` for those GPU features, although they are only supported on Linux and Windows.

Found the cause. I had implemented the optional mirroring in the linter, but not in the fixer. 🤦 

Fixed in 129e18bb83f03048b2cfc451d333a124c7ef3a70.

**Review by caugner — APPROVED** [review]

**Pull request merged**